src/classes/RD2_NamingService.cls

```diff
@@ -39,11 +39,19 @@ public with sharing class RD2_NamingService {
     private static final String CURRENCY_ISO_CODE_FIELD = UTIL_Currency.CURRENCY_ISO_CODE_FIELD;
     public static final Integer MAX_NAME_LENGTH = 80;
 
+    /**
+    * @description Support special Naming Tokens that can be used in the RD Name
+    */
+    private static final Map<String, String> namingTokens = new Map<String, String>{
+        '$RECURRING_NAME_SUFFIX$' => System.Label.RecurringDonationNameSuffix,
+        '$RECURRING_OBJECT_NAME$' => UTIL_Describe.getObjectLabel('npe03__Recurring_Donation__c')
+    };
+
     /***
-    * @description RD Naming format options 
+    * @description Supported RD Naming format options
     */
-    public static final String RD_NAMING_OPTION_1 = '{!npe03__Donor_Name__c} {!npe03__Amount__c} - \"Recurring\"';
-    public static final String RD_NAMING_OPTION_2 = '{!npe03__Donor_Name__c} {!npe03__Installment_Period__c} Recurring Donation - {!npe03__Amount__c}';
+    public static final String RD_NAMING_OPTION_1 = '{!npe03__Donor_Name__c} {!npe03__Amount__c} - $RECURRING_NAME_SUFFIX$';
```

Comment:
@andrewyu-salesforce FYI -- I changed the text in these to use a token that we can replace more easily in code. You can use that same replace logic using the `namingTokens` map above to create the label to render in settings picklist, while storing these versions in the custom settings field.